fasjsonclient/clientbuilder/patch.py

```diff
@@ -0,0 +1,47 @@
+import functools
+
+import urllib3
+import six
+import gssapi
+import requests
+from requests_gssapi import HTTPSPNEGOAuth, REQUIRED
+
+from fasjsonclient.openapi_client.rest import RESTClientObject
+
+
+def request(self, method, url, query_params=None, headers=None,
+                body=None, post_params=None, _preload_content=True,
+                _request_timeout=None, **kwargs):
+  """Patch fucntion to use requests + gssapi integration instead of urllib3"""
+  if query_params:
+    url += f'?{urlencode(query_params)}'
+  principal_name = kwargs.get('principal_name', 'admin@EXAMPLE.TEST')
```

Comment:
Same thing here, I think it would be better for the function to fail if the `principal_name` argument is not provided, rather than succeed with a testing value.